jetty-core/jetty-security/src/main/java/org/eclipse/jetty/security/SecurityHandler.java

```diff
@@ -0,0 +1,761 @@
+
+    /**
+     * @return the list of discovered authenticatorFactories
+     */
+    public List<Authenticator.Factory> getKnownAuthenticatorFactories()
+    {
+        return __knownAuthenticatorFactories;
+    }
+
+    /**
+     * @return the realmName
+     */
+    @Override
+    public String getRealmName()
+    {
+        return _realmName;
+    }
+
+    /**
+     * @param realmName the realmName to set
+     * @throws IllegalStateException if the SecurityHandler is running
+     */
+    public void setRealmName(String realmName)
+    {
+        if (isRunning())
+            throw new IllegalStateException("running");
+        _realmName = realmName;
+    }
+
+    /**
+     * @return the authMethod
+     */
+    @Override
+    public String getAuthMethod()
+    {
+        return _authMethod;
+    }
+
+    /**
+     * @param authMethod the authMethod to set
+     * @throws IllegalStateException if the SecurityHandler is running
+     */
+    public void setAuthMethod(String authMethod)
+    {
+        if (isRunning())
+            throw new IllegalStateException("running");
+        _authMethod = authMethod;
+        if (_authenticator != null && !_authenticator.getAuthMethod().equals(_authMethod))
+            _authenticator = null;
+    }
+
+    @Override
+    public String getParameter(String key)
+    {
+        return _parameters.get(key);
+    }
+
+    @Override
+    public Set<String> getParameterNames()
+    {
+        return _parameters.keySet();
+    }
+
+    /**
+     * Set an authentication parameter for retrieval via {@link AuthConfiguration#getParameter(String)}
+     *
+     * @param key the key
+     * @param value the init value
+     * @return previous value
+     * @throws IllegalStateException if the SecurityHandler is started
+     */
+    public String setParameter(String key, String value)
+    {
+        if (isStarted())
+            throw new IllegalStateException("started");
+        return _parameters.put(key, value);
+    }
+
+    protected LoginService findLoginService() throws Exception
+    {
+        java.util.Collection<LoginService> list = getServer().getBeans(LoginService.class);
+        LoginService service = null;
+        String realm = getRealmName();
+        if (realm != null)
+        {
+            for (LoginService s : list)
+            {
+                if (s.getName() != null && s.getName().equals(realm))
+                {
+                    service = s;
+                    break;
+                }
+            }
+        }
+        else if (list.size() == 1)
+            service = list.iterator().next();
+
+        return service;
+    }
+
+    protected IdentityService findIdentityService()
+    {
+        return getServer().getBean(IdentityService.class);
+    }
+
+    @Override
+    protected void doStart()
+        throws Exception
+    {
+        // complicated resolution of login and identity service to handle
+        // many different ways these can be constructed and injected.
```

Comment:
Yeah, too complicated.
Eventually it all boils down to 1 Authenticator, which needs 1 LoginService and 1 IdentityService. 
After doStart() is called, these don't change (they certainly cannot in the middle of handling a request, despite the setters), so I'd guess `ServletHandler -> Authenticator -> LoginService -> IdentityService` is enough?

No need for beans to lookup on the server, no need for setters repeated in multiple interfaces, no need for double check that all these objects are actually consistent otherwise throw ISE, etc.

Reply:
yeah this is all existing code.  I will look at how we actually do it in our modules/xml and simplify to support just that.

Reply:
As discussed, leaving this as is